src/GenericFactory.sol

```diff
@@ -63,8 +68,10 @@ contract GenericFactory is IGenericFactory, Owned {
             address lPointer = lByteCode[i];
             uint256 lSize = lPointer.code.length - 0x01;
 
-            // TODO: Go check/annotate all asm for memory safety.
-            assembly {
+            // SAFETY:
+            // This assembly block is memory safe because it writes to a region already
+            // allocated via the previous assembly block
+            assembly ("memory-safe") {
```

Comment:
When we release back to solidity between assembly blocks, the memory model must be correct. We clearly copy `lSize` into memory and then release back to solidity, which is not safe. We need to write this entire loop in assembly, or write `0x40` slot in between iterations. Given gas efficiency is not a huge worry, simplest fix would be to update the free memory pointer before returning back to solidity.

Reply:
okay so I fixed it by updating the free memory pointer after `extcodecopy`, and also updated the docs to say that the free memory pointer has been updated, thus it's safe.